### PR TITLE
Add validation for max length on string columns with user input

### PR DESCRIPTION
**Pull request by sonalkr132**
Rails before 4.2 use to set max limit of 255 in DB for columns with `string` type (`character varying(255)`). This means in existing prod/staging db we have this limit set on all string columns except `required_rubygems_version` (added after 4.2 change).
I am not sure what are we suppose to do about missing `limit: 255` in schema.rb. If someone runs migration today, they won't get the limits set in existing prod DB. I guess we could generate a redundant migration which just sets these limits on string columns?

### CODE REVIEW

File: app/models/version.rb [hunk 1]
```diff
@@ -16,11 +15,11 @@ class Version < ApplicationRecord
   serialize :licenses
   serialize :requirements
 
-  validates :number,   format: { with: /\A#{Gem::Version::VERSION_PATTERN}\z/ }
-  validates :platform, format: { with: Rubygem::NAME_PATTERN }
+  validates :number, length: { maximum: Gemcutter::MAX_FIELD_LENGTH }, format: { with: /\A#{Gem::Version::VERSION_PATTERN}\z/ }
+  validates :platform, length: { maximum: Gemcutter::MAX_FIELD_LENGTH }, format: { with: Rubygem::NAME_PATTERN }
   validates :full_name, presence: true, uniqueness: { case_sensitive: false }
   validates :rubygem, presence: true
-  validates :required_rubygems_version, length: { minimum: 1, maximum: MAX_FIELD_LENGTH }, allow_blank: true
```
Comment (sonalkr132) [inline app/models/version.rb:23]: Let me know if `minimum: 1` was covering a case I am missing. As far as I can tell, this is not needed.

File: test/unit/version_test.rb [hunk 2]
```diff
@@ -730,33 +748,37 @@ class VersionTest < ActiveSupport::TestCase
...
+        @version.metadata = { "home" => "http://github.com/bestgemever" }
+        assert @version.validate
+        assert_equal @version.errors.messages[:metadata], []
       end
 
-      should "be invalid with maxiumum length" do
-        assert_raise ActiveRecord::RecordInvalid do
-          @spec.metadata = { "home" => ("hello" * 2000) }
-          @version.update_attributes_from_gem_specification!(@spec)
-        end
```
Comment (sonalkr132) [inline test/unit/version_test.rb:759]: changed this to check for error message explicitly as existing test was raising invalid record for home url being invalid (not max length).

**Comment by rietta** [comment]
You mentioned `limit 255` in schema.rb. When I added a migration to my typo squat PR at https://github.com/rubygems/rubygems.org/pull/2341/files#diff-1acd2e7e27a227829d5d14a91c863bb6, notice that the schema dump after the migration ran added those. I've added validations for string length to my Rails applications for years. I think it is a very sensible thing to do, especially with PostgreSQL that will raise an exception for input longer than the varchar length rather than truncating it.

File: app/models/version.rb (hunk 1 above)
Comment (rietta) [reply app/models/version.rb:23]: I presume it was trying to say that the version is present and not blank.

Comment (rietta) [reply app/models/version.rb:23]: `allow_blank: true` is contradictory with minimum length 1.

**Pull request merged**